pkg/controller/controller_binding.go

```diff
@@ -434,7 +475,7 @@ func (c *controller) reconcileServiceInstanceCredential(binding *v1alpha1.Servic
 	// that's actually a soft delete-- i.e. we have some finalization to do.
 	if finalizers := sets.NewString(binding.Finalizers...); finalizers.Has(v1alpha1.FinalizerServiceCatalog) {
```

Comment:
This logic works, but I think we should add a clause to this if statement that makes it clear that we might be doing this for orphan mitigation, not just finalization.